vectordraw.py

```diff
@@ -48,7 +70,41 @@ def check_tip_x(check, vectors):
 def check_tip_y(check, vectors):
     vec = vectors[check['vector']]
     if _check_coordinate(check, vec.tip.y):
-        return 'Vector {} does not tip at correct point.'.format(vec.name)
+        return 'Vector {} does not end at correct point.'.format(vec.name)
+
+def _check_coords(vec, expected, tolerance):
+    for expected_coords, vec_coords in ([expected[0], vec.tail], [expected[1], vec.tip]):
+        if expected_coords[0] == '_':
+            if expected_coords[1] != '_':
+                # Only y given; not interested in x.
+                if abs(expected_coords[1] - vec_coords.y) > tolerance:
+                    return True
+        elif expected_coords[1] == '_':
+            # Only x given; not interested in y.
+            if abs(expected_coords[0] - vec_coords.x) > tolerance:
+                return True
```

Comment:
@mtyaka most likely it could be achieved in a more elegant way

    def check_coord(expected, actual):
        return expected == '_' or abs(expected - actual) > tolerance
    
    if not (check_coord(expected_coords[0], vec_coords.x) and check_coord(expected_coords[1], vec_coords.x):
        return True  # do I understand right that this function returns True if check fails?

    if expected_coords[0] != '_' and expected_coords[1] != '_':
        # same as below

The trick is, distance check is a stronger assumption, so if it holds above checks will always hold as well. Basically, checking for both coordinates be in `tolerance` distance of target point independently forms a square; checking for distance from target point forms an inscribed circle.